#131246955 Authenticated users can view game history

adetola-ralph opened the pull request:
#### What does this PR do?

Adds feature that allows authenticated users to view their game history.
#### Description of Task to be completed?
- A user should be able to click a button which directs to the history page
- A user should see their previous games played on the history page
#### How should this be manually tested?

After login, the user would see the `history` link as part of the top menu on the main page, on the link being clicked the user should be redirected to a history page that displays the users history.
#### What are the relevant pivotal tracker stories?
#131246955 Users should be able to view their personal game history
#### Screenshots (if appropriate)

![project-kinsan-1](https://cloud.githubusercontent.com/assets/20815290/19084639/4b31e904-8a5f-11e6-9aa5-7b15a97d6786.png)
![project-kinsan-2](https://cloud.githubusercontent.com/assets/20815290/19084640/4b330546-8a5f-11e6-93b2-10afe138b1ed.png)

Commit:
Feature(base-files): Add base files needed to for the frontend

Start #131246955

Commit:
Feature(base-files): Add base files needed to for the frontend

Start #131246955

Commit:
Feature(base-files): Add base files needed to for the frontend

Start #131246955

Commit:
Feature(add-history-route): Add /#!/history route to the main controller

Commit:
Feature(add-history-route): Add /#!/history route to the main controller

Commit:
Feature(add-history-route): Add /#!/history route to the main controller

Commit:
Feature(history-test): Add test files for angular history code

Add spec files to test the angular js code to be written for
getting authenticated user's game history. The test files will
contain tests for an angular factory(service) and angular
controller.

Commit:
Feature(history-test): Add test files for angular history code

Add spec files to test the angular js code to be written for
getting authenticated user's game history. The test files will
contain tests for an angular factory(service) and angular
controller.

Commit:
Feature(history-test): Add test files for angular history code

Add spec files to test the angular js code to be written for
getting authenticated user's game history. The test files will
contain tests for an angular factory(service) and angular
controller.

Commit:
Feature(history-test): Add test files for angular history code

Add spec files to test the angular js code to be written for
getting authenticated user's game history. The test files will
contain tests for an angular factory(service) and angular
controller.

Commit:
Feature(service-test): Add spec to test History service is defined

Add some specs to the history.test.js file to test if
the History service is defined. This is just a basic
test to ascertain the existence of the existence of the
service.

Commit:
Feature(service-test): Add spec to test History service is defined

Add some specs to the history.test.js file to test if
the History service is defined. This is just a basic
test to ascertain the existence of the existence of the
service.

Commit:
Feature(service-test): Add spec to test History service is defined

Add some specs to the history.test.js file to test if
the History service is defined. This is just a basic
test to ascertain the existence of the existence of the
service.

Commit:
Feature(test-files): Add location of new files to karma

The new files created for the History service and History
controller have been added to the karma config file. Also,
the History test file has also been added.

Commit:
Feature(test-files): Add location of new files to karma

The new files created for the History service and History
controller have been added to the karma config file. Also,
the History test file has also been added.

Commit:
Feature(test-files): Add location of new files to karma

The new files created for the History service and History
controller have been added to the karma config file. Also,
the History test file has also been added.

Commit:
Feature(pass-test): Add History service to pass the first test

Commit:
Feature(pass-test): Add History service to pass the first test

Commit:
Feature(pass-test): Add History service to pass the first test

Commit:
Feature(karma-config): Add all js files in app folder to karma

Commit:
Feature(karma-config): Add all js files in app folder to karma

Commit:
Feature(karma-config): Add all js files in app folder to karma

Commit:
Feature(service-test): Add spec to test history service has getHistory function

Commit:
Feature(service-test): Add spec to test history service has getHistory function

Commit:
Feature(service-test): Add spec to test history service has getHistory function

demiladebamgbose committed:
add history api endpoints

demiladebamgbose committed:
add history api endpoints

demiladebamgbose committed:
add history api endpoints

demiladebamgbose committed:
add history api endpoints

Commit:
Refactor(standards) simplify code by multi-lining

Commit:
Refactor(standards) simplify code by multi-lining

Commit:
Merge branch 'feature/#131246955/user-can-view-game-history' of github.com:andela/project-kinsan into feature/#131246955/user-can-view-game-history

Commit:
Feature(add-history-controller): Add new angular module `services.history` to main module `mean`

Add services.History as a dependency to the mean angular module which serves as the main controller. Also registers services.History controller and its dependency

Commit:
Feature(add-history-controller): Add new angular module `services.history` to main module `mean`

Add services.History as a dependency to the mean angular module which serves as the main controller. Also registers services.History controller and its dependency

Commit:
Test(History-serice-test): Add test for HistoryFac test

add tests to check that the HistoryFac service exists and if it
has the getHistory function defined. It also checks if the getHistory
function makes a http get request to /api/users/:id/history endpoint

Commit:
Test(History-serice-test): Add test for HistoryFac test

add tests to check that the HistoryFac service exists and if it
has the getHistory function defined. It also checks if the getHistory
function makes a http get request to /api/users/:id/history endpoint

Commit:
Feature(Implement-getHistory): Implement getHistory() of HistoryFac

implement getHistory() function in HistoryFac service to make a
http request to /api/users/:id/history to get user history

Commit:
Feature(Implement-getHistory): Implement getHistory() of HistoryFac

implement getHistory() function in HistoryFac service to make a
http request to /api/users/:id/history to get user history

Commit:
Test(history-controller-test): Add test for HistoryController

add test to check if HistoryController controller exists and if it
calls the getHistory() function of HistoryFac service

Commit:
Test(history-controller-test): Add test for HistoryController

add test to check if HistoryController controller exists and if it
calls the getHistory() function of HistoryFac service

Commit:
Feature(history-controller): Implement HistoryController

implement HistoryController to call the getHistory() function of the
HistoryFac service.

Commit:
Feature(history-controller): Implement HistoryController

implement HistoryController to call the getHistory() function of the
HistoryFac service.

Commit:
Feature(add-new-files): Add new controller and service file to food.jade

Add HistoryController and HistoryFac files to footer.jade file

Commit:
Feature(add-new-files): Add new controller and service file to food.jade

Add HistoryController and HistoryFac files to footer.jade file

Commit:
Refactor(variable-name): Change variable name for HistoryController in test file

Commit:
Refactor(variable-name): Change variable name for HistoryController in test file

Commit:
Feature(check-authentication): Check HistoryController if user is authenticated

Commit:
Feature(check-authentication): Check HistoryController if user is authenticated

Commit:
Feature(html-template): Add content to history.tpl.html

Add content to the template that displays the history of an
authenticated user. The page shows a message to unauthenticated
users indicating that they are not logged in. Also shows a message
to users who don't have a game history yet

Commit:
Feature(html-template): Add content to history.tpl.html

Add content to the template that displays the history of an
authenticated user. The page shows a message to unauthenticated
users indicating that they are not logged in. Also shows a message
to users who don't have a game history yet

Commit:
Feature(add-history-link): Add link to history in the top menu

add history link to the top menu. The link only shows up for
authenticated users

Commit:
Feature(add-history-link): Add link to history in the top menu

add history link to the top menu. The link only shows up for
authenticated users

Commit:
Feature(change-binding): Change binding on history.tpl.html

changed from {{}} binding to using ng-bind online on line 59 of
history.tpl.html

Commit:
Feature(change-binding): Change binding on history.tpl.html

changed from {{}} binding to using ng-bind online on line 59 of
history.tpl.html

Commit:
Refactor(const-to-var): Change const to var on line 6 in history.ctr.js

Commit:
Refactor(const-to-var): Change const to var on line 6 in history.ctr.js

Commit:
Refactor(const-to-var): Change let to var on line 144 in history.test.js

Commit:
Refactor(const-to-var): Change let to var on line 144 in history.test.js

Commit:
Refactor(refactor-history-test): Rewrite history.test.js to remove Hound CI errors

Commit:
Refactor(refactor-history-test): Rewrite history.test.js to remove Hound CI errors

Commit:
#130341343 signup with jwt (#10)

* chore(readme-update): update read me with new relevant info
Finishes #130373385

* feature(signup): JWT for signup

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in auth.js, routes.js, all.js and signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in auth.js and signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): JWT for signup

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in auth.js, routes.js, all.js and signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in auth.js and signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt
- Allow signup without the need of a token
Finishes #130341343

* feature(signup): signup-with-jwt
- Rebase with develop branch
Finishes #130341343

* feature(signup): signup-with-jwt
- Rebase with develop branch
Finishes #130341343

* feature(signup): JWT for signup

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in auth.js, routes.js, all.js and signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in auth.js and signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt

-Update code in signup-spec.js
to meet hound specifications.

Finishes #130341343

* feature(signup): signup-with-jwt
- Allow user signup without token
Finishes #130341343

* feature(signup): signup-with-jwt
- Rebase with develop
Finishes #130341343

* feature(signup): signup-with-jwt
- Rebase with develop
Finishes #130341343

* feature(signup): signup-with-jwt
- Add password hashing function to signup
Finishes #130341343

ruqoyyasadiq commented on app/controllers/history.js:
```diff
@@ -0,0 +1,30 @@
+  History.findOne({gameId: req.body.gameId}, function (err, history) {
+    if (history) {
+      res.json(history);
+    }
```
why not test for if history doesn't exist then create history, rather than checking if history exists. something like:

```
if(!history) {
  createHistoryMethod()
}
```

ruqoyyasadiq commented on test/server/history-api.test.js:
```diff
@@ -0,0 +1,77 @@
+      ],
+      'completed': true,
+      'winner': '57e96f012206d8884e000001'
+    };
```
Have a `beforeEach()` block to create test history data. Also, have an `afterEach` block to drop the data after running the test, to avoid errors in the when re-running the tests again.

Commit:
Feature(delete-history): Add a function to the history controller to delete history from the db

Commit:
Feature(delete-history): Add a function to the history controller to delete history from the db

Commit:
Feature(delete-endpoint): Add route for history delete endpoint

Commit:
Feature(delete-endpoint): Add route for history delete endpoint

Commit:
Feature(tests): Add afterEach to delete added history after each /api/history spec runs

Commit:
Feature(tests): Add afterEach to delete added history after each /api/history spec runs

Commit:
Feature(jasmine-jquery): Install jasmine-jquery from bower

Commit:
Feature(jasmine-jquery): Install jasmine-jquery from bower

Commit:
Refactor(move-response): Move response in history file to its file

move the mock response json object to its own file and require it through the getJSONFixture of jasmine-jquery

Commit:
Refactor(move-response): Move response in history file to its file

move the mock response json object to its own file and require it through the getJSONFixture of jasmine-jquery

Commit:
Merge branch 'feature/#131246955/user-can-view-game-history' of github.com:andela/project-kinsan into feature/#131246955/user-can-view-game-history

ruqoyyasadiq reviewed (approved):
Approved!
